src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -209,5 +207,33 @@ static void RegisterMessageRoute(Type mappedType, string address, Dictionary<Typ
             }
             routeTableEntries[mappedType] = new RouteTableEntry(mappedType, route);
         }
+
+        static List<Type> GetBaseTypes(Type messageType, Conventions conventions)
+        {
+            var result = new List<Type>();
+            var baseType = messageType.BaseType;
```

Comment:
Is it enough to walk the base types of do we need to look at implemented interfaces as well?